meraki_wireless.tf

```diff
@@ -627,46 +627,43 @@ resource "meraki_wireless_ssid_identity_psk" "networks_wireless_ssids_identity_p
+            ranges = try(length(wireless_ssid.unavailability_schedules.ranges) == 0, true) ? null : [
+              for range in try(wireless_ssid.unavailability_schedules.ranges, []) : {
+                start_day  = try(range.start_day, local.defaults.meraki.domains.organizations.networks.wireless.ssids.unavailability_schedules.ranges.start_day, null)
+                start_time = try(range.start_time, local.defaults.meraki.domains.organizations.networks.wireless.ssids.unavailability_schedules.ranges.start_time, null)
+                end_day    = try(range.end_day, local.defaults.meraki.domains.organizations.networks.wireless.ssids.unavailability_schedules.ranges.end_day, null)
+                end_time   = try(range.end_time, local.defaults.meraki.domains.organizations.networks.wireless.ssids.unavailability_schedules.ranges.end_time, null)
               }
             ]
-            ranges_in_seconds = try(length(wireless_ssid.schedules.ranges_in_seconds) == 0, true) ? null : [
-              for ranges_in_second in try(wireless_ssid.schedules.ranges_in_seconds, []) : {
-                start = try(ranges_in_second.start, local.defaults.meraki.domains.organizations.networks.wireless.ssids.schedules.ranges_in_seconds.start, null)
-                end   = try(ranges_in_second.end, local.defaults.meraki.domains.organizations.networks.wireless.ssids.schedules.ranges_in_seconds.end, null)
+            ranges_in_seconds = try(length(wireless_ssid.unavailability_schedules.ranges_in_seconds) == 0, true) ? null : [
+              for ranges_in_second in try(wireless_ssid.unavailability_schedules.ranges_in_seconds, []) : {
+                start = try(ranges_in_second.start, local.defaults.meraki.domains.organizations.networks.wireless.ssids.unavailability_schedules.ranges_in_seconds.start, null)
+                end   = try(ranges_in_second.end, local.defaults.meraki.domains.organizations.networks.wireless.ssids.unavailability_schedules.ranges_in_seconds.end, null)
               }
             ]
-          } if try(wireless_ssid.schedules, null) != null
+          } if try(wireless_ssid.unavailability_schedules, null) != null
         ]
       ]
     ]
   ])
 }
 
-resource "meraki_wireless_ssid_schedules" "networks_wireless_ssids_schedules" {
-  for_each          = { for v in local.networks_wireless_ssids_schedules : v.key => v }
+resource "meraki_wireless_ssid_schedules" "networks_wireless_ssids_unavailability_schedules" {
+  for_each          = { for v in local.networks_wireless_ssids_unavailability_schedules : v.key => v }
   network_id        = each.value.network_id
   number            = each.value.number
   enabled           = each.value.enabled
   ranges            = each.value.ranges
   ranges_in_seconds = each.value.ranges_in_seconds
-  depends_on = [
-    meraki_wireless_ssid.networks_wireless_ssids
-  ]
```

Comment:
Note (not a change request): This removal was unexected, but I think it is fine - we already refer to `meraki_wireless_ssid.networks_wireless_ssids[...].number` in `number`.

This would only be an issue if we had to ensure all SSIDs (not just the one the particular schedule is for) were processed first, for which I don't see any evidence: the original commit that added the `depends_on` (for many of these resources at the same time) (https://github.com/netascode/terraform-meraki-nac-meraki/commit/0766463258c9091bd4ee178acf55d0474354bdcb#diff-5ef594e0ac1a28c42b2215d674de7f5f8b70a2cbcb4b168827c930f1b058780bR447) has no reason mentioned in the message.